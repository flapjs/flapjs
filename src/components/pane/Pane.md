# Pane

## Motivation
Cause I love all of you. <3

## Work Log
- Created Pane component.