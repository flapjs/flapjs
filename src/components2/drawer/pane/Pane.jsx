import React from 'react';
import PropTypes from 'prop-types';
import Style from './Pane.module.css';

import { TinyDownIcon } from '@flapjs/components2/icons/Icons.js';

/**
 * A React component that can do anything you want :D.
 */
class Pane extends React.Component
{
    constructor(props)
    {
        super(props);

        this.state = {
            open: true
        };

        this.onHeaderClick = this.onHeaderClick.bind(this);
    }

    onHeaderClick(e)
    {
        this.setState(prev =>
        {
            return {
                open: !prev.open
            };
        });
    }

    /** @override */
    render()
    {
        const props = this.props;
        const state = this.state;

        const open = state.open;
        
        return (
            <section className={Style.container
                + ' ' + (props.className || '')
                + (open ? ' open': '')}>
                <header>
                    <button className={Style.dropdown} onClick={this.onHeaderClick}>
                        <span className={Style.title}>
                            {props.title}
                        </span>
                        <TinyDownIcon className="icon" />
                    </button>
                </header>
                <main>
                    {props.children}
                </main>
            </section>
        );
    }
}

Pane.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node
    ]),
};
Pane.defaultProps = {
};

export default Pane;
