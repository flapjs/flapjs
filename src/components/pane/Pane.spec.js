import React from 'react';
import { shallow } from 'enzyme';

import Pane from './Pane.jsx';

let wrapper;

beforeEach(() =>
{
    wrapper = shallow(<Pane/>);
});

describe('<Pane/> rendering', () =>
{
    test('renders correctly', () =>
    {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<Pane/> interaction', () =>
{
    /** I promise I will test you soon... */
});

describe('<Pane/> lifecycle', () =>
{
    /** ...later. */
});
