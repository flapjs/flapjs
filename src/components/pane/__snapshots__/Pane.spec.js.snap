// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pane/> rendering renders correctly 1`] = `
<section
  className="container  open"
>
  <header>
    <button
      className="dropdown"
      onClick={[Function]}
    >
      <span
        className="title"
      />
      <ForwardRef
        className="icon"
      />
    </button>
  </header>
  <main />
</section>
`;
